Include/cpython/genobject.h

```diff
@@ -35,7 +35,7 @@ PyAPI_FUNC(PyObject *) PyCoro_New(PyFrameObject *,
 
 /* --- Asynchronous Generators -------------------------------------------- */
 
-typedef struct _PyAsyncGenObject PyAsyncGenObject;
+typedef PyGenObject PyAsyncGenObject;
```

Comment:
PyCoroObject and PyAsyncGenObject are mentioned in the docs, so I think we need to leave this even though we're not using these names.